gobblin-service/src/main/java/org/apache/gobblin/service/modules/flowgraph/pathfinder/AbstractPathFinder.java

```diff
@@ -155,6 +159,26 @@
     this.destDatasetDescriptor = DatasetDescriptorUtils.constructDatasetDescriptor(destDatasetDescriptorConfig);
   }
 
+  static List<String> getDataNodes(Config flowConfig, String identifierKey, Config config) {
+    List<String> dataNodes = ConfigUtils.getStringList(flowConfig, identifierKey);
+    return dataNodes.stream().map(dataNode -> convertDataNode(dataNode, config)).collect(Collectors.toList());
+  }
+
+  private static String getDataNode(Config flowConfig, String identifierKey, Config config) {
+    String dataNode = ConfigUtils.getString(flowConfig, identifierKey, "");
+    return convertDataNode(dataNode, config);
+  }
+
+  protected static String convertDataNode(String dataNode, Config config) {
```

Comment:
nit: semantically, this is not node 'conversion', but node ID aliasing.  suggest: more accurate terminology